Fix telegram chats

azogue opened the pull request:
## Description:

Related issue (if applicable): fixes #7694 

Fix negative `chat_id`s used in Telegram groups:
- Negative `chat_id`s for groups.
- Include `chat_id` in event data.
- Handle KeyError when receiving other types of messages, as `new_chat_member` ones, and send them as text.

I know this is like a duplicate of #7688, but this has more changes to allow **receiving** messages from a group, not only sending them.

It's tested and intended to be included in 0.45.1

azogue committed:
bugfix for Telegram chat_ids

- Negative `chat_id`s for groups.
- Include `chat_id` in event data.
- Handle KeyError when receiving other types of messages, as `new_chat_member` ones, and send them as text.

azogue committed:
unused import

MartinHjelmare commented on homeassistant/components/telegram_bot/__init__.py:
```diff
@@ -503,12 +505,18 @@ def process_message(self, data):
             else:
-                event_data[ATTR_TEXT] = data[ATTR_TEXT]
+                # Some other thing...
+                _LOGGER.warning('SOME OTHER THING RECEIVED --> "%s"', data)
```
Clarify to something like:
```py
_LOGGER.warning("Message without text data received: %s", data)
```
Note the use of quotes. That's home assistant standard now.

MartinHjelmare reviewed (changes requested):
Just a minor comment.

azogue committed:
fix double quote style, fix boolean expr, change warning msg

azogue committed:
mistake

azogue committed:
some more fixes

- fix if condition for msg bad fields.
- return True for a correct but not allowed or not recognized message: if not, the message arrives continuously.
- Allow to receive messages from unauthorized users if they come from authorized groups.

azogue committed:
support for `edited_message`s

- They come as normal messages, except for the 'edited_message' field instead of 'message'.

arsaboo commented:
Is this related https://github.com/home-assistant/home-assistant/issues/7696 ?

azogue commented:
@arsaboo, not exactly, #7696 is because the changes of 0.45 imply that [we now have to define a `telegram_bot`](https://home-assistant.io/components/notify.telegram/). This was not tagged as a breaking change ...

MartinHjelmare commented:
@sander76 after the refactor of #7454, telegram notify requires telegram_bot to be setup. So they are no longer unrelated.

michaelarz commented:
Hi! it seems to be broken again. negative chat_id calls this:
2017-06-19 18:23:56 ERROR (MainThread) [homeassistant.components.telegram_bot] Incoming message is not allowed ({'from': {'last_name': 'Last', 'first_name': 'First', 'id': 2823xxxx, 'language_code': 'en'}, 'message_id': 1963, 'text': '/ping', 'entities': [{'offset': 0, 'type': 'bot_command', 'length': 5}], 'chat': {'title': 'Title', 'all_members_are_administrators': True, 'id': -17616yyyy, 'type': 'group'}, 'date': 1497889436})
2017-06-19 18:24:04 WARNING (Thread-6) [homeassistant.components.telegram_bot] Unallowed targets: [-17616yyyy], using default: 2823xxxx

azogue commented:
Hi @michaelarz, have you included the `-17616yyy` chat_id to the list of allowed chat ids in the `telegram_bot` config? Could you show us your yaml config, please?

michaelarz commented:
Hi @azogue, thank you, that was exatly what I forgot to do. Sorry, my fault.
